http.go

```diff
@@ -270,6 +272,22 @@ func (hc *HttpCall) ResponseBodyContains(value string) Step {
 	})
 }
 
+// ResponseBodyMatches is a Step that when executed ensures there is
+// a non-nil hc.ResponseBody and errors unless the hc.ResponseBody
+// matches the regular expression parameter.
+func (hc *HttpCall) ResponseBodyMatches(pattern string) Step {
+	return NewNamedStep("ResponseBodyMatches", func() error {
+		if err := hc.ReceiveBody(); err != nil {
+			return err
+		}
+		var regex = regexp.MustCompile(pattern)
```

Comment:
I would also prefer that the whole body is a single if-elseif-else block in keeping with the vast majority of the other steps here.